orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/appengine/AppEngineServerGroupCreator.groovy

```diff
@@ -75,8 +77,10 @@ class AppEngineServerGroupCreator implements ServerGroupCreator {
       List<ArtifactAccountPair> configArtifacts = operation.configArtifacts
       if (configArtifacts != null && configArtifacts.size() > 0) {
         operation.configArtifacts = configArtifacts.collect { artifactAccountPair ->
-          def artifact = artifactUtils.getBoundArtifactForStage(stage, artifactAccountPair.id, artifactAccountPair.artifact)
-          artifact.artifactAccount = artifactAccountPair.account
+          def artifact = artifactUtils.getBoundArtifactForStage(stage, artifactAccountPair.id, objectMapper.convertValue(artifactAccountPair.artifact, Artifact.class))
+          if (artifactAccountPair.account != null){
```

Comment:
I'm curious about the need for this change here (which seems unrelated)...did you encounter a bug where this was set even when null?

This seems reasonable, but for a change on a release branch I'd want the minimal change that fixes the bug.

Reply:
@ezimanyi:
 I added this logic:
`if (artifactAccountPair.account != null) { ... }`
Because in this branch, it breaks the code. The `account` is already set in the Artifact object, and the assignment of account in Artifact was actually setting the account to `null` instead of keeping the value already set in Artifact.

The master branch actually already has this fix, when it calls `ArtifactUtil.withAccount()`, which does the check for null, instead of blindly setting it.